### PR TITLE
Migration guide for component theme update

### PR DESCRIPTION
**Pull request by QuncCccccc**
This PR is to add a migration guide page for the change in `CardTheme`, `DialogTheme` and `TabBarTheme` in the material library

## Presubmit checklist

- [x] This PR is marked as draft with an explanation if not meant to land until a future stable release.
- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

File: src/content/release/breaking-changes/component-theme-normalization.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
+---
+title: Component Theme Normalization
+description: >-
+  `CardTheme`, `DialogTheme`, and `TabBarTheme` has been normalized to follow
```
Comment (Piinks) [inline src/content/release/breaking-changes/component-theme-normalization.md:4]:
```suggestion
  `CardTheme`, `DialogTheme`, and `TabBarTheme` have been normalized to follow
```

File: src/content/release/breaking-changes/component-theme-normalization.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  the Flutter's convention for component themes.
```
Comment (Piinks) [inline src/content/release/breaking-changes/component-theme-normalization.md:5]:
```suggestion
  Flutter's convention for component themes.
```

File: src/content/release/breaking-changes/component-theme-normalization.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+---
+
+## Summary
+
+`CardTheme`, `DialogTheme` and `TabBarTheme` have been refactored to make them
+conform to Flutter's conventions for component themes. `CardThemeData`,
```
Comment (Piinks) [inline src/content/release/breaking-changes/component-theme-normalization.md:11]:
```suggestion
`CardTheme`, `DialogTheme` and `TabBarTheme` were refactored to 
conform to Flutter's conventions for component themes. `CardThemeData`, 
```

File: src/content/release/breaking-changes/component-theme-normalization.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  the Flutter's convention for component themes.
+---
+
+## Summary
+
+`CardTheme`, `DialogTheme` and `TabBarTheme` have been refactored to make them
+conform to Flutter's conventions for component themes. `CardThemeData`, 
+`DialogThemeData` and `TabBarThemeData` were added to define overrides for the
+defaults of the component visual properties. Future releases will continue to
+normalize component themes like these for more consistent theming experience in
```
Comment (Piinks) [inline src/content/release/breaking-changes/component-theme-normalization.md:14]:
```suggestion
normalize component themes like these for a more consistent theming experience in
```

**Review by Piinks — APPROVED** [review]
LGTM! Thanks @QuncCccccc!

File: src/content/release/breaking-changes/component-theme-normalization.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
+---
+title: Component Theme Normalization
```
Comment (sfshaza2) [inline src/content/release/breaking-changes/component-theme-normalization.md:2]:
```suggestion
title: Component theme normalization
```

File: src/content/release/breaking-changes/index.md [hunk 6]
```diff
@@ -57,12 +55,16 @@ release, and listed in alphabetical order:
 * [Stop generating `AssetManifest.json`][]
 * [Deprecate `TextField.canRequestFocus`][]
 * [Set default for SystemUiMode to Edge-to-Edge][]
+* [Material 3 Tokens Update in Flutter][]
+* [Component Theme Normalization][]
```
Comment (sfshaza2) [inline src/content/release/breaking-changes/index.md:59]:
```suggestion
* [Component theme normalization][]
```

**Review by sfshaza2 — APPROVED** [review]
lgtm

**Comment by esodot** [comment]
thanks, found a typo as well, even if already close :)

The return type of the **compoenent** theme xTheme.of() method and Theme.of().xTheme now has also changed to xThemeData accordingly.

**Pull request merged**